File: 1435-xor-queries-of-a-subarray/1435-xor-queries-of-a-subarray.java
class Solution {
    public int[] xorQueries(int[] arr, int[][] queries) {
        int[] ans = new int[queries.length];
        for (int i = 1; i < arr.length; i++) arr[i] ^= arr[i - 1];
        int idx = 0;
        for (int[] q: queries) {
            if (q[0] == 0) ans[idx] = arr[q[1]];
            else ans[idx] = arr[q[0] - 1] ^ arr[q[1]];
            idx++;
        }
        return ans;
    }
}

File: 1435-xor-queries-of-a-subarray/1435-xor-queries-of-a-subarray.kt
class Solution {
    fun xorQueries(arr: IntArray, queries: Array<IntArray>): IntArray {
        val ref: IntArray = arr.runningReduce { acc, curr -> acc xor curr }.toIntArray();
        val ans: IntArray = IntArray(queries.size) { 0 };
        var idx: Int = 0;
        for (q in queries) 
            ans[idx++] = if(q[0] == 0) ref[q[1]] else ref[q[0] - 1] xor ref[q[1]];
        return ans;
    }
}